### PR TITLE
1280 unhandled keycloak log in failure

### PR DESCRIPTION
**Pull request by Derstilon**
- [X] https://github.com/yaptide/ui/issues/1280
- [ ] https://github.com/yaptide/ui/issues/1282
- [X] https://github.com/yaptide/ui/issues/1127

### CODE REVIEW

**Comment by grzanka** [comment]
Still when trying to run a simulation locally I get:
image

**Comment by grzanka** [comment]
@Derstilon tests are failing now

image

**Comment by grzanka** [comment]
I am able to run simulations now, it seems that just the tests needs to be fixed

**Comment by grzanka** [comment]
@Derstilon can you fix the conflicts as well ?

File: src/ThreeEditor/js/Storage.js [hunk 1]
```diff
@@ -39,27 +39,52 @@ function Storage() {
...
+				const transaction = database.transaction(['states'], 'readwrite');
+				const objectStore = transaction.objectStore('states');
+				const request = objectStore.get(0);
+				request.onsuccess = function (event) {
+					callback(event.target.result);
+				};
+			} catch (error) {
+				console.error(error);
+				callback();
+			}
```
Comment (ostatni5) [inline src/ThreeEditor/js/Storage.js:69]: Examine if `try/catch` is needed.

File: src/ThreeEditor/js/Storage.js [hunk 2]
```diff
@@ -39,27 +39,52 @@ function Storage() {
...
+				const request = objectStore.put(data, 0);
+				request.onsuccess = function () {
+					devLog(
+						'Saved state to IndexedDB.',
+						`${(performance.now() - start).toFixed(2)}ms`
+					);
+				};
+			} catch (error) {
+				console.error(error);
+			}
```
Comment (ostatni5) [inline src/ThreeEditor/js/Storage.js:87]: Examine if `try/catch` is needed.

File: src/WrapperApp/components/Simulation/SimulationCard.tsx [hunk 3]
```diff
@@ -427,7 +429,11 @@ export default function SimulationCard({
 						color='info'
 						size='small'
 						onClick={onClickSaveToFile}
-						disabled={!Boolean(simulationStatus.jobState === StatusState.COMPLETED)}>
+						disabled={
+							!Boolean(
```
Comment (ostatni5) [inline src/WrapperApp/components/Simulation/SimulationCard.tsx:433]: **nitpick:** You can probably remove the Boolean keyword

File: src/WrapperApp/components/Simulation/SimulationCardGrid.tsx [hunk 4]
```diff
@@ -222,11 +225,15 @@ export function PaginatedSimulationsFromBackend({
 					variant='contained'
 					color='info'
 					startIcon={<QueuePlayNextIcon />}
-					disabled={!isBackendAlive}
+					disabled={!Boolean(isBackendAlive && yaptideEditor)}
```
Comment (ostatni5) [inline src/WrapperApp/components/Simulation/SimulationCardGrid.tsx:228]: Boolean

File: src/services/AuthService.tsx [hunk 5]
```diff
@@ -168,67 +170,82 @@ const Auth = ({ children }: GenericContextProviderProps) => {
 	}, [reachServer]);
 
 	useEffect(() => {
-		if (user !== null && user.source !== 'keycloak' && isServerReachable)
+		if (!demoMode && user?.source !== 'keycloak' && isServerReachable)
 			setRefreshInterval(prev => (prev === undefined ? 3000 : prev));
 		// 3 seconds in ms default interval for refresh when logged in with username and password
 		else if (!isServerReachable || !user?.source) setRefreshInterval(undefined);
```
Comment (ostatni5) [inline src/services/AuthService.tsx:176]: If can be omitted

File: src/services/AuthService.tsx [hunk 6]
```diff
@@ -168,67 +170,82 @@ const Auth = ({ children }: GenericContextProviderProps) => {
...
+		/**
+		 * TODO: Check if user is authorized to use this application (e.g. by checking if user is in a certain group)
+		 */
+		const validUser = true;
+
+		if (!validUser)
+			openRejectKeycloakDialog({
+				reason: 'You are not authorized to use this application.',
+				keycloakAuth: { keycloak, initialized }
 			});
```
Comment (ostatni5) [inline src/services/AuthService.tsx:192]: Add resolve/reject

File: src/services/AuthService.tsx [hunk 7]
```diff
@@ -168,67 +170,82 @@ const Auth = ({ children }: GenericContextProviderProps) => {
...
 	useEffect(() => {
-		if (user !== null && user.source !== 'keycloak' && isServerReachable)
+		if (!demoMode && user?.source !== 'keycloak' && isServerReachable)
 			setRefreshInterval(prev => (prev === undefined ? 3000 : prev));
 		// 3 seconds in ms default interval for refresh when logged in with username and password
 		else if (!isServerReachable || !user?.source) setRefreshInterval(undefined);
-	}, [isServerReachable, user]);
+	}, [demoMode, isServerReachable, user]);
 
 	const tokenVerification = useCallback(() => {
```
Comment (ostatni5) [inline src/services/AuthService.tsx:179]: check if you must return smth

File: src/services/KeycloakAuthService.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+const KeycloakAuth = ({ children }: GenericContextProviderProps) => {
+	const { altAuth } = useConfig();
+
+	return altAuth ? (
+		<KeycloakProvider
+			client={authInstance}
+			initOptions={initOptions}>
+			<KeycloakAuthContextProvider
+				value={{
```
Comment (ostatni5) [inline src/services/KeycloakAuthService.tsx:43]: extract to var

**Review by ostatni5 — CHANGES_REQUESTED** [review]

**Comment by grzanka** [comment]
Logging works , the too frequent calls to keycloak disappeared

**Comment by grzanka** [comment]
I have issues with submitting simulations to batch, 
this is an error from backend
```
INFO:werkzeug:172.18.0.5 - - [25/Oct/2023 14:34:13] "POST /jobs/batch HTTP/1.0" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2213, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2193, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 298, in error_router
    return original_handler(e)
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 2190, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1486, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 298, in error_router
    return original_handler(e)
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1484, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.10/site-packages/flask/app.py", line 1469, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 489, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask/views.py", line 109, in view
    return current_app.ensure_sync(self.dispatch_request)(**kwargs)
  File "/usr/local/lib/python3.10/site-packages/flask_restful/__init__.py", line 604, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/usr/local/app/yaptide/routes/utils/decorators.py", line 24, in wrapper
    return f(user, *args, **kwargs)
  File "/usr/local/app/yaptide/routes/batch_routes.py", line 78, in post
    result = submit_job(payload_dict=payload_dict, files_dict=input_dict["input_files"], user=user,
  File "/usr/local/app/yaptide/batch/batch_methods.py", line 77, in submit_job
    array_id = int(line.split()[-1])
ValueError: invalid literal for int() with base 10: 'id:'
```

this is JSON send by frontend:
```
{"input_json":{"metadata":{"version":"0.11","type":"Editor","generator":"YaptideEditor.toJSON"},"project":{"title":"Proton pencil beam in water","description":"Proton pencil beam with initial kinetic energy 150 MeV and 1% energy spread.\nThe target is a cylindrical water phantom with a radius of 10 cm and a length of 20 cm.\n\nWe score dose from all particles and fluence according to primary particles (protons), secondary protons, and all protons.","viewManager":{"ViewPanelXY":{"cameraMatrix":[1,0,0,0,0,0.9999999999994964,9.999999999621397e-7,0,0,-9.999999999621397e-7,0.9999999999994964,0,0,-0.00009999999999621423,99.99999999994992,1],"clipPlane":{"planeConstant":0,"visible":true}},"ViewPanel3D":{"cameraMatrix":[0.7071067811865477,5.5511151231257815e-17,-0.707106781186547,0,-0.4082482904638628,0.8164965809277258,-0.4082482904638629,0,0.5773502691896248,0.5773502691896252,0.5773502691896255,0,10,10,10.000000000000002,1]},"ViewPanelY":{"cameraMatrix":[5.551115123125587e-16,-4.4408920985004704e-16,0.9999999999999649,0,0.9999999999994634,9.999999996956521e-7,-6.661338147750695e-16,0,-9.999999998066744e-7,0.9999999999994634,3.3306690738753473e-16,0,-0.00009999999999998334,99.99999999994999,6.1232339957357454e-21,1],"clipPlane":{"planeConstant":0,"visible":true}},"ViewPanelX":{"cameraMatrix":[1.1102230246251573e-16,1.0000000000000007,2.2204460492503146e-16,0,9.999999995846643e-7,-4.4408920985006163e-16,0.9999999999994978,0,0.9999999999994978,-2.2204460492503082e-16,-9.999999996956866e-7,0,99.99999999994995,0,-0.0000999999999999833,1],"clipPlane":{"planeConstant":0,"visible":true}},"ViewPanel":{"cameraMatrix":[0.07426714888874009,0,0.9972383820310657,0,0.1890359579236641,0.9818692455756362,-0.014078039800121761,0,-0.9791576978239105,0.18955944870338465,0.07292062945044075,0,-27.694761919291807,5.361550864645869,2.0625068629119263,1]}},"history":{"undos":[],"redos":[]}},"figureManager":{"uuid":"0F0F0F0F-0F0F-0F0F-0F0F-0F0F0F0F0F0F","name":"Figure Manager","type":"FigureManager","metadata":{"version":"0.11","type":"Manager","generator":"FigureManager.toJSON"},"figures":[{"name":"Water_phantom_cylinder","type":"CylinderFigure","uuid":"7ab62ba9-6dba-4543-b50b-ce31e143d482","visible":false,"geometryData":{"geometryType":"HollowCylinderGeometry","position":[0,0,10],"rotation":[0,0,0],"parameters":{"innerRadius":0,"radius":5,"depth":20}},"colorHex":0},{"name":"Vacuum_cylinder","type":"CylinderFigure","uuid":"aeeffad9-ed86-44ae-bfe4-b5f78580741b","visible":false,"geometryData":{"geometryType":"HollowCylinderGeometry","position":[0,0,10.5],"rotation":[0,0,0],"parameters":{"innerRadius":0,"radius":5.5,"depth":22}},"colorHex":0}]},"zoneManager":{"uuid":"368269D3-6F9B-4241-84A0-5D908D10E196","name":"Zones","type":"ZoneManager","metadata":{"version":"0.11","type":"Manager","generator":"ZoneManager.toJSON"},"zones":[{"uuid":"9ee041c1-acc0-433f-bc66-90e057b72474","name":"Water_phantom_zone","type":"BooleanZone","visible":true,"materialUuid":"49ce56b9-ee48-4640-88d4-bcfd660d8c0e","materialPropertiesOverrides":{},"unionOperations":[[{"mode":"union","objectUuid":"7ab62ba9-6dba-4543-b50b-ce31e143d482"}]],"subscribedObjects":{"7ab62ba9-6dba-4543-b50b-ce31e143d482":1}},{"uuid":"a2c614cf-7074-4abe-a757-4a850f6731b3","name":"Vacuum_zone","type":"BooleanZone","visible":true,"materialUuid":"cd7c2eb0-e91f-4144-b4eb-536e4626cc09","materialPropertiesOverrides":{},"unionOperations":[[{"mode":"union","objectUuid":"aeeffad9-ed86-44ae-bfe4-b5f78580741b"},{"mode":"subtraction","objectUuid":"7ab62ba9-6dba-4543-b50b-ce31e143d482"}]],"subscribedObjects":{"7ab62ba9-6dba-4543-b50b-ce31e143d482":1,"aeeffad9-ed86-44ae-bfe4-b5f78580741b":1}}],"worldZone":{"uuid":"73e69243-fc89-4f8f-b353-466e9d3921df","type":"WorldZone","name":"World Zone","marginMultiplier":1.1,"autoCalculate":false,"materialUuid":"1dd2e8f0-6684-42c2-ad35-7aa55358204e","visible":false,"geometryData":{"geometryType":"HollowCylinderGeometry","position":[0,0,10.5],"rotation":[0,0,0],"parameters":{"innerRadius":6,"radius":6,"depth":24}}}},"detectorManager":{"uuid":"9A7D17B8-BAE4-441F-965F-490CAA617EC6","name":"DetectManager","type":"DetectorManager","metadata":{"version":"0.11","type":"Manager","generator":"DetectorManager.toJSON"},"detectors":[{"name":"CylZ_Mesh","type":"Detector","uuid":"186ec65a-8511-418b-8218-8890138c6b12","visible":true,"colorHex":327424,"geometryData":{"geometryType":"Cyl","position":[0,0,10],"rotation":[0,0,0],"parameters":{"radius":5,"innerRadius":0,"depth":20,"radialSegments":1,"zSegments":400}}},{"name":"YZ_Mesh","type":"Detector","uuid":"3306e98b-5da2-46ee-8892-ac0191d645a7","visible":true,"colorHex":65535,"geometryData":{"geometryType":"Mesh","position":[0,0,10],"rotation":[0,0,0],"parameters":{"width":0.5,"height":4,"depth":20,"xSegments":1,"ySegments":80,"zSegments":400}}},{"name":"EntrySlab","type":"Detector","uuid":"53fff3c4-30de-4210-9066-20dc0f90cccb","visible":true,"colorHex":65535,"geometryData":{"geometryType":"Cyl","position":[0,0,0.05],"rotation":[0,0,0],"parameters":{"radius":5,"innerRadius":0,"depth":0.1,"radialSegments":1,"zSegments":1}}},{"name":"PeakSlab","type":"Detector","uuid":"deb6a8f8-fab3-488e-8a01-9bb3e8ea4649","visible":true,"colorHex":65535,"geometryData":{"geometryType":"Cyl","position":[0,0,15.35],"rotation":[0,0,0],"parameters":{"radius":5,"innerRadius":0,"depth":0.1,"radialSegments":1,"zSegments":1}}}]},"specialComponentsManager":{"uuid":"0F0F0F0F-0F0F-0F0F-0F0F-0F0F0F0F0FFF","name":"Special Components","type":"SpecialComponentManager","metadata":{"version":"0.11","type":"Manager","generator":"SpecialComponentManager.toJSON"}},"materialManager":{"uuid":"0F0F0F0F-0F0F-0F0F-0F0F-0F0F0FFFFFFF","name":"Material Manager","type":"MaterialManager","materials":[{"uuid":"1dd2e8f0-6684-42c2-ad35-7aa55358204e","sanitizedName":"black_hole","name":"BLACK HOLE","icru":0,"density":1,"color":196613},{"uuid":"49ce56b9-ee48-4640-88d4-bcfd660d8c0e","sanitizedName":"water_liquid","name":"WATER, LIQUID","icru":276,"density":1,"color":34047,"opacity":0.2,"transparent":true},{"uuid":"cd7c2eb0-e91f-4144-b4eb-536e4626cc09","sanitizedName":"vacuum","name":"VACUUM","icru":1000,"density":1,"color":16119285,"opacity":0.1,"transparent":true}],"selectedMaterials":{"49ce56b9-ee48-4640-88d4-bcfd660d8c0e":22,"cd7c2eb0-e91f-4144-b4eb-536e4626cc09":1,"1dd2e8f0-6684-42c2-ad35-7aa55358204e":1},"metadata":{"version":"0.11","type":"Manager","generator":"MaterialManager.toJSON"}},"scoringManager":{"name":"Outputs","type":"ScoringManager","uuid":"D0F67F59-F311-4F8E-A06F-5A07D618FDE2","outputs":[{"name":"z_profile","type":"Output","uuid":"d24dd7eb-6a7c-40d0-b39e-f4424b3dd937","quantities":[{"uuid":"3b0bc80c-5d78-4899-870f-23167e99fe30","name":"Dose","type":"Quantity","keyword":"Dose","modifiers":[]},{"uuid":"53ba1036-4975-4a96-a6bd-3d336fc72612","name":"FluenceProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"Fluence","modifiers":[]},{"uuid":"050f3919-a59b-45a4-9326-38d788c16ab7","name":"FluencePrimaries","type":"Quantity","filter":"eacf5223-2411-4529-8abe-2c0fb2daf57f","keyword":"Fluence","modifiers":[]},{"uuid":"b3286437-b2a1-4ee9-b7da-8cc2cc435fcd","name":"FluenceSecondaryProtons","type":"Quantity","filter":"6ba215e8-359e-447d-bac9-4f2808faed51","keyword":"Fluence","modifiers":[]}],"detectorUuid":"186ec65a-8511-418b-8218-8890138c6b12","trace":false},{"name":"yz_profile","type":"Output","uuid":"6e69ee1b-a73c-4936-a5a2-e81f5bf8bc04","quantities":[{"uuid":"828d1276-a60e-4e31-9da7-88c84587445a","name":"Dose","type":"Quantity","keyword":"Dose","modifiers":[]},{"uuid":"be97ed10-07af-49d2-b91a-5bfe09c0c228","name":"FluenceProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"Fluence","modifiers":[]},{"uuid":"69305c71-de72-466c-86bc-ba5e58635a9e","name":"FluenceSecondaryProtons","type":"Quantity","filter":"6ba215e8-359e-447d-bac9-4f2808faed51","keyword":"Fluence","modifiers":[]}],"detectorUuid":"3306e98b-5da2-46ee-8892-ac0191d645a7","trace":false},{"name":"entrance","type":"Output","uuid":"2e46a6ab-35d7-4fce-9de8-406a47fc9180","quantities":[{"uuid":"9730c840-7163-4125-8596-7d26735c373a","name":"Dose","type":"Quantity","keyword":"Dose","modifiers":[]},{"uuid":"48b2c7a4-58c4-4ff7-816b-865ce146ae95","name":"KineticEnergyPrimaries","type":"Quantity","filter":"eacf5223-2411-4529-8abe-2c0fb2daf57f","keyword":"AvgEnergy","modifiers":[]},{"uuid":"a9164784-1770-48d5-a772-3342fb06efe9","name":"KineticEnergyProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"AvgEnergy","modifiers":[]},{"uuid":"2bf3b68d-a855-443a-8ea0-2bf05eeb9e65","name":"KineticEnergySecondaryProtons","type":"Quantity","filter":"6ba215e8-359e-447d-bac9-4f2808faed51","keyword":"AvgEnergy","modifiers":[]},{"uuid":"d0c4addd-0775-420e-9240-d5ba3e279a2b","name":"dLETProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"dLET","modifiers":[]},{"uuid":"a4e9cac1-a643-4448-bc4a-89977101cc67","name":"tLETProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"tLET","modifiers":[]},{"uuid":"de64c12b-15cc-4e72-9262-b5507a738c9f","name":"ProtonsKineticEnergySpectrum","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"Fluence","modifiers":[{"diffType":"E","lowerLimit":0,"upperLimit":160,"binsNumber":640,"isLog":false,"uuid":"1afaf3d8-443f-4bac-8999-be914ff0691a"}]}],"detectorUuid":"53fff3c4-30de-4210-9066-20dc0f90cccb","trace":false},{"name":"peak","type":"Output","uuid":"a9e8737f-edae-45fe-9395-1e6b220ef384","quantities":[{"uuid":"d13a9fc5-d13c-4e5a-9fc4-decfe15c9ee4","name":"Dose","type":"Quantity","keyword":"Dose","modifiers":[]},{"uuid":"54b85511-c885-41bd-a81d-a97842197764","name":"KineticEnergyPrimaries","type":"Quantity","filter":"eacf5223-2411-4529-8abe-2c0fb2daf57f","keyword":"AvgEnergy","modifiers":[]},{"uuid":"e7ff88d1-e46a-4a69-b676-e8e3ed1e67fd","name":"KineticEnergyProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"AvgEnergy","modifiers":[]},{"uuid":"95c292b9-966e-42fe-91a0-3d0f0db55876","name":"KineticEnergySecondaryProtons","type":"Quantity","filter":"6ba215e8-359e-447d-bac9-4f2808faed51","keyword":"AvgEnergy","modifiers":[]},{"uuid":"efd9f82c-8e0e-4d3d-a22d-70a0cc11ef7b","name":"dLETProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"dLET","modifiers":[]},{"uuid":"dac16bbd-e361-46d3-8eae-9c69571aea96","name":"tLETProtons","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"tLET","modifiers":[]},{"uuid":"65f821c8-a654-4f6f-b666-0fdc8c925cf5","name":"ProtonsKineticEnergySpectrum","type":"Quantity","filter":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","keyword":"Fluence","modifiers":[{"diffType":"E","lowerLimit":0,"upperLimit":160,"binsNumber":640,"isLog":false,"uuid":"07a5db58-a8ea-4175-93e7-9fe74ca14d34"}]}],"detectorUuid":"deb6a8f8-fab3-488e-8a01-9bb3e8ea4649","trace":false}],"filters":[{"uuid":"9d0bae09-9ebe-46fe-9e59-bd7888db9442","name":"Protons","type":"Filter","rules":[{"uuid":"e5f50132-6d2a-4606-a58d-24a764d1a413","keyword":"Z","operator":"==","value":1},{"uuid":"f25fb8bb-fdaa-49ae-9483-76bf6f8de0a6","keyword":"A","operator":"==","value":1}]},{"uuid":"eacf5223-2411-4529-8abe-2c0fb2daf57f","name":"Primaries","type":"Filter","rules":[{"uuid":"624050ff-9761-4688-99e3-b3ba9592ae78","keyword":"Z","operator":"==","value":1},{"uuid":"6cef8b17-9337-46e5-b522-70a9d1b26324","keyword":"A","operator":"==","value":1},{"uuid":"d9b7c82e-338f-40d9-b316-67ed0bab2eb1","keyword":"GEN","operator":"==","value":0}]},{"uuid":"6ba215e8-359e-447d-bac9-4f2808faed51","name":"Secondary_protons","type":"Filter","rules":[{"uuid":"3d485d37-4753-4afa-85ab-ed0f9bc5ae3b","keyword":"Z","operator":"==","value":1},{"uuid":"b9b5c2a9-6625-409f-81e2-827799cef0a5","keyword":"A","operator":"==","value":1},{"uuid":"919e4280-15ad-449b-9495-1cf1ac0337bc","keyword":"GEN","operator":">=","value":1}]}],"metadata":{"version":"0.11","type":"Manager","generator":"ScoringManager.toJSON"}},"beam":{"name":"Beam","type":"Beam","uuid":"0F0F0F0F-0F0F-0F0F-0F0F-0F0F0F0FFFFF","position":[0,0,0],"direction":[0,0,1],"energy":150,"energySpread":1.5,"energyLowCutoff":0,"energyHighCutoff":1000,"sigma":{"type":"Gaussian","x":0,"y":0},"sad":{"type":"none","x":0,"y":0},"divergence":{"distanceToFocal":0,"x":0,"y":0},"particle":{"a":1,"id":2,"z":1},"colorHex":16776960,"numberOfParticles":10000,"sourceFile":{"name":"","value":""},"sourceType":"simple"},"physic":{"energyLoss":0.03,"enableNuclearReactions":true,"energyModelStraggling":"Vavilov","multipleScattering":"Moliere","stoppingPowerTable":"ICRU91","availableStoppingPowerFiles":{}},"hash":"b71689509f5f3da39b077bf0724d29b377317fd4"},"input_type":"editor","ntasks":30,"sim_type":"shieldhit","title":"Proton pencil beam in water","batch_options":{"cluster_name":"default","array_header":"","array_options":{},"collect_header":"","collect_options":{}}}
```

**Review by ostatni5 — APPROVED** [review]

**Review by grzanka — APPROVED** [review]

**Pull request merged**